dask/array/fft.py

```diff
@@ -146,10 +146,8 @@ def fft_wrap(fft_func, kind=None, dtype=None):
     >>> parallel_ifft = dff.fft_wrap(np.fft.ifft)
     """
     if scipy is not None:
-        if fft_func is scipy.fftpack.rfft:
-            raise ValueError("SciPy's `rfft` doesn't match the NumPy API.")
-        elif fft_func is scipy.fftpack.irfft:
-            raise ValueError("SciPy's `irfft` doesn't match the NumPy API.")
+        if fft_func.__module__.startswith("scipy.fftpack"):
```

Comment:
Why this change?

Reply:
Thanks for reviewing @phofl .

If we ignore the norm argument, only `rfft` and `irfft` were considered not to match NumPy's API. However, if we do consider the norm argument (as in this PR), then none of the functions from `scipy.fftpack` (including `fft`, `ifft`, `fft2`, etc.) match NumPy's API. 

This changes raises an error for all those functions from the depracated `scipy.fftpack` (including `rfft` and `irfft`).

Reply:
got it, thanks. Can you include the function that is called in the error message to make it more descriptive?

Reply:
> we ignore the norm argument, only rfft and irfft were considered not to match NumPy's API. However, if we do consider the norm argument (as in this PR), then none of the functions from scipy.fftpack (including fft, ifft, fft2, etc.) match NumPy's API.

Can we keep backwards compatibility somehow? Dask supports older version of scipy as well

Reply:
1.7.2, did it start raising deprecation warnings back then?

Reply:
Keeping backward compatibility would mean to replicate the behavior of `norm={“backward”, “ortho”, “forward”}` of `scipy.fft` functions as a patch for `scipy.fftpack` calls.

Reply:
> did it start raising deprecation warnings back then?

It doesn't raise deprecation warnings - it is just marked as legacy in the docs.